src/main/webapp/script.js

```diff
@@ -0,0 +1,382 @@
+
+    // draw new image on it, scaling the image to fit in canvas
+    ctx.drawImage(image, 0, 0, canvas.width, canvas.height);
+  };
+  prepareCanvas('input-canvas');
+  prepareCanvas('output-canvas');
+
+  // remove download button
+  const downloadButton = document.getElementById('download-button');
+  if (downloadButton !== null) {
+    downloadButton.remove();
+  }
+}
+
+/**
+ * Class for a rectangle to blur
+ */
+class Rect {
+  /**
+   * @param {function} func Can be min or max function
+   * @param {string} property Property which all the items
+   *                          in arrayObject have
+   * @param {Array} arrayObject array of Objects
+   * @return {Number} min or max (depending on func) value of the
+   *         specified property among arrayObject items.
+   */
+  static get(func, property, arrayObject) {
+    let ans = arrayObject[0][property];
+
+    for (const item of arrayObject) {
+      ans = func(ans, item[property]);
+    }
+
+    return ans;
+  }
+
+  /**
+   * @constructor
+   * @param {JsonRect} _rect
+   * Constructs Rect from rectangle object we get from json
+   * Rectange object we get from json has 4 points - its corners,
+   * our Rect has top left corner, width and height.
+   */
+  constructor(_rect) {
+    this.leftX = Rect.get(Math.min, 'x', _rect);
+    this.topY = Rect.get(Math.min, 'y', _rect);
+    this.width = Rect.get(Math.max, 'x', _rect) - this.leftX;
+    this.height = Rect.get(Math.max, 'y', _rect) - this.topY;
+  }
+}
+
+/**
+ * Function that blurs specific rectangles of an image
+ * @param {Array<JsonRect>} rectsToBlur A list of coordinates of
+ *                                      rectangles to blur
+ * @param {Image} image Image on which we want to blur those rectangles
+ * @return {HTMLCanvasElement} canvas with an image with blurred rectangles
+ * The general idea is to create canvas with blurred image on it,
+ * and canvas with not blurred image on it. Then we need to extract
+ * rectangles from blurred image and put them on top of not-blurred image.
+ * The image is blurred using Gaussian blur algorithm
+ * (it is implemented in canvas.filter.blur)
+ */
+function getBlurredImage(rectsToBlur, image) {
+  // create result canvas and draw not-blurred image on it
+  const resultCanvas = createCanvasForImage(image);
+  const resultCtx = resultCanvas.getContext('2d');
+  resultCtx.drawImage(image, 0, 0);
+
+  // create canvas to put blurred areas there
+  const blurredRectsCanvas = createCanvasForImage(image);
+  const blurredRectsCtx = blurredRectsCanvas.getContext('2d');
+
+  blurredRectsCtx.globalCompositeOperation = 'destination-in';
+
+  for (const _rect of rectsToBlur) {
+    const rect = new Rect(_rect);
+
+    // blur radius depends on the rect size
+    const blurRadius = getBlurRadius(rect);
+    console.log(blurRadius);
+
+    // we will get blurred rectangle from this canvas
+    const hiddenBlurredCanvas = getBlurredCanvas(image, blurRadius);
```

Comment:
I think blurring the whole image again for every rectangle is not the best idea. Maybe store in an outer variable the blurredCanvas corresponding to a blurRadius and check if it was already created before creating a new one?

That wouldn't work if blurRadius would be different to every rectangle size (like a formula to calculate blurRadius from the rect size), but it works in current case where there are only 2 blurRadius possible.

Reply:
I was planning to make blurRadius different for most of the rectangles in the upcoming version, that is why I didn't want to store blurred canvases now.
I will include this optimisation in TODO for the experiments pull request.